new/missing-method-missing.md

```diff
@@ -0,0 +1,156 @@
+---
+RFC: unassigned
+Author: John Keiser <john@johnkeiser.com>
+Status: Draft
+Type: Standards Track
+---
+
+# Remove magic and method_missing from Chef::Resource lookup
+
+This proposal:
+
+- Switches to explicit methods for Chef recipe DSL instead of method_missing
+- Removes all special treatment of classes in the Chef::Resource and Chef::Provider namespaces
+- Automatically adds resource DSL for all named descendants of Chef::Resource, no matter what their namespace
```

Comment:
As far as the everyone vs. no one decision, here are my thoughts:

1. I agree with you from a cleanliness standpoint: everyone should call `requires`
2. From a practical standpoint this would make a lot of resources stop working
3. I think that, as you have proposed in the past, the real Resource base should be a mixin that we will create later.  That is the point at which we get to do things that really cut backcompat deeply, because it won't affect existing Chef::Resource derived stuff.